webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm

```diff
@@ -444,33 +450,56 @@ tabContent model details =
             let
               groupTreeElem : Group -> Html Msg
               groupTreeElem item =
-                    li [class "jstree-node jstree-leaf"]
-                    [ i[class "jstree-icon jstree-ocl"][]
-                    , a[href "#", class "jstree-anchor"]
-                      [ i [class "jstree-icon jstree-themeicon fa fa-sitemap jstree-themeicon-custom"][]
-                      , span [class "treeGroupName tooltipable"][text item.name, (if item.dynamic then (small [class "greyscala"][text "- Dynamic"]) else (text ""))]
-                      , div [class "treeActions-container"]
-                        [ span [class "treeActions"][ span [class "tooltipable fa action-icon accept", onClick (SelectGroup (NodeGroupId item.id) True)][]]
-                        , span [class "treeActions"][ span [class "tooltipable fa action-icon except", onClick (SelectGroup (NodeGroupId item.id) False)][]]
-                        ]
+                let
+                  checkIncludeOrExclude : List RuleTarget -> Bool
+                  checkIncludeOrExclude lst =
+                    let
+                      getTargetIds : List RuleTarget -> List String
+                      getTargetIds listTargets =
+                        List.concatMap (\target -> case target of
+                           NodeGroupId groupId -> [groupId]
+                           Composition i e -> getTargetIds [i, e]
+                           Special spe -> [spe]
+                           Node node -> [node]
+                           Or t  -> getTargetIds t
+                           And t -> getTargetIds t
```

Comment:
And is more complicated (or is okay) and you need to keep targets that are member of all targets list in t

But you may not have to do it right know never mind (maybe put a comment here that And is not ok as of now)